libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/RX/NetworkInterface.c

```diff
@@ -168,20 +261,35 @@ static void prvEMACDeferredInterruptHandlerTask( void *pvParameters )
 
     uint8_t *buffer_pointer;
 
+    /* Some variables related to monitoring the PHY. */
+    TimeOut_t xPhyTime;
+    TickType_t xPhyRemTime;
+    const TickType_t ulMaxBlockTime = pdMS_TO_TICKS( 100UL );
+
+    vTaskSetTimeOutState( &xPhyTime );
+    xPhyRemTime = pdMS_TO_TICKS( PHY_LS_LOW_CHECK_TIME_MS );
+
+    configPRINTF( ( "Deferred Interrupt Handler Task started\n" ) );
+    xTaskToNotify = ether_receive_check_task_handle;
+
     for( ;; )
     {
-        xTaskToNotify = ether_receive_check_task_handle;
+        prvMonitorResources();
```

Comment:
Seems like the monitoring should be conditionally compiled, so that it may be omitted from production code.

Reply:
You are right about that. In a release version, you don't want logging that nobody looks at.

I propose to make it conditional on `ipconfigHAS_PRINTF`. In a release version, that macro will be undefined or `0`.
More than 50% of the issues reported on the forum can be resolved by just monitoring these three resources. So I think it's important to have it by default, just like `ipconfigHAS_PRINTF` is `1` by default.

Reply:
Just committed: "Monitor resources only if FreeRTOS_printf is defined"
The function `prvMonitorResources()` will only be declared and called if `ipconfigHAS_PRINTF` is defined as `1`.